EIPS/eip-future_salaries.md

```diff
@@ -0,0 +1,71 @@
+
+2. This would effectively decrease ETH supply. The necessity to decrease the ETH supply (by locking ETH in Future Salaries system for a long time) is a well-known important things to be done.
+
+> For a multi-asset smart contract platform to function as a store of value, proper incentives must be put in place to align in the growth in value of a network's assets with its underlying security. Or put another way, the platform's native token must be a good value capture of the platform's aggregate asset value. If the intrinsic value of a platform's native token is limited to transaction fee payment, its value would be determined solely by transaction demand, instead of the demand of asset storage. (https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0001-positioning/0001-positioning.md)
+
+In other words, we need to measure the ETH "real" value. This EIP proposes to allocate *much* (it probably may be more than 90%) of miner's ETH fees to Future Salaries contract. So the cost of ETH will be associated with the real ("job") value of the conditional tokens of this contract.
+
+So, this proposal partly solves both the well-known problems of
+- decreasing ETH supply
+- making ETH cost proportionally increasing when tokens' value increases (in other word incresing the ETH intristic value)
+
+As the above mentioned RFC explains, it is good for miners/validators, because:
+- It reduces manipulation incentives such as selfish mining
+- It highly increases the security of the Ethereum network
+- It increases the ETH intristic value, so miners/validators spend less on energy per amount of money earned.
+- There is the danger (https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0001-positioning/0001-positioning.md) that without radical measures like this proposal the value of ETH (not of Ethereum tokens) will become zero in some future, so greatly downshifting miners/validators.
+
+## Specification
+1. Finish the smart contract audit (it was already paid for) of [SalaryWithDAO](https://github.com/vporton/future-contracts/blob/master/contracts/SalaryWithDAO.sol) and [DefaultDAOInterface](https://github.com/vporton/future-contracts/blob/master/contracts/DefaultDAOInterface.sol) contracts.
+2. After the audit is finished, deploy these contracts (together with a DAOstack DAO controlling the `DefaultDAOInterface` with voting tokens initially allocated by a community consensus) and create an oracle ID using the `SalaryWithDAO` API.
+3. Audit and deploy [DonateETH](https://github.com/vporton/donations/blob/main/contracts/DonateETH.sol) redirecting ETH payments to this oracle, too.
+4. Discuss the fractions of miners fees and transfer fees of ETH to be transferred to `DonateETH` at every mining and transfer operation.
+5. Change the Ethereum clients to do these ETH transfers.
```

Comment:
The specification should be a technical specification that explains how to change the core consensus protocol.  This isn't an appropriate place for a todo list.  EIPs should be written in their final form, and if some information is not yet ready at draft time just leave it out or put in a `TBD` or similar placeholder.

Reply:
I don't quite understand what is `TBD` placeholder. Where is this file? How to make my text a `TBD` placeholder?

I will remove "Finish the smart contract audit (it was already paid for)", but it seems appropriate for me to ask the community for an audit of `DonateETH` because audit is not a part of the proposal itself but a part of its rationale verification. Is it OK?

Do you insist that I need to list particular Ethereum addresses of contracts? If I specify only the exact has of the contracts source and the version of the Solidity compiler proposed, it is quite exact, except that the address of the contract is not specified in the advance. Would this be OK? It makes little sense to redeploy several times, let us save some gas, OK?

Reply:
I mean just put the word `TBD` in for stuff you haven't figured out fully yet.
For example:
```
## Backward Compatibility
TBD
```
It is a way to get the structure of the EIP done before you have finished figuring out all of the pieces.  Ideally, the specification section should have *some* substantive content before merging as a draft, but it doesn't have to be complete.

> I will remove "Finish the smart contract audit (it was already paid for)", but it seems appropriate for me to ask the community for an audit of DonateETH because audit is not a part of the proposal itself but a part of its rationale verification. Is it OK?

The EIP process is purely a technical specification and standardization process.  It has nothing to do with governance, solicitation, argumentation, essays, etc.  The EIP should *only* describe the technical specification you are proposing and you should refer to it *elsewhere* when discussing things like funding, auditing processes, debating why it should be included, etc.

Since this is a `Core` EIP, changes to consensus are what this should focus on.  If a contract baked into Ethereum is necessary for this EIP to make sense that can be included, and it should (ideally) be deployed at a predetermined address on all networks.  The address doesn't need to be filled in until the very end, before Last Call, so you are free to iterate on the contract and just have something like :
```
## Specification
`CONTRACT_ADDRESS` = `TBD`

Prior to `FORK_BLOCK_NUMBER`, the following contract will be deployed to the network and exist at address `CONTRACT_ADDRESS`....
```